kuma/core/models.py

```diff
@@ -48,12 +49,19 @@ class IPBan(models.Model):
     def delete(self, *args, **kwargs):
         self.deleted = timezone.now()
         self.save()
-        IPBanJob().invalidate(self.ip)
 
     def __unicode__(self):
         return u'%s banned on %s' % (self.ip, self.created)
 
 
+def invalidate_ipban_caches(sender, instance, **kwargs):
+    IPBanJob().invalidate(instance.ip)
+
+
+models.signals.post_save.connect(invalidate_ipban_caches, sender=IPBan)
+models.signals.post_delete.connect(invalidate_ipban_caches, sender=IPBan)
```

Comment:
I'm not sure if `pre_delete` wouldn't be better here since the `post_delete` works under the assumption that the table row is already gone and only the python ORM object has the info needed. To prevent future mistakes that extend the `invalidate_ipban_caches` signal handler somehow I'd rather go with that.

Also you should use the `receiver` decorator if you're importing it, like so:

```
@receiver(models.signals.post_save, sender=IPBan)
@receiver(models.signals.pre_delete, sender=IPBan)
def invalidate_ipban_caches(sender, instance, **kwargs):
    IPBanJob().invalidate(instance.ip)
```